### ISSUE TITLE
Frame Rate Drop

### ISSUE THREAD

**Issue by mserour** [ouster-lidar/ouster-ros#137]
I have an OS1-64 running using ROS2 Foxy Driver, Ubuntu 20.04 that I received yesterday. My machine is quite powerful since it has the latest processor from Intel I9-13900K and I am using the standard launch file command to run the LIDAR. The LIDAR starts to consistently publish the data at 10 FPS, however, with time evolves, the FPS keeps dropping and there is some sort of skittering/discontinuity in the LIDAR visualization from RVIZ. 

When we tried to update the resolution of the output LIDAR image to 512x10, the rate was more consistent, however, it isn't that consistent over time (observed by hz the topic). Also one more observation, if we covered the LIDAR i.e. no points, the rate is consistent since there are no points. I believe it has something to do with the processing done to output the LIDAR image.

Any recommendations?

Thank you.

**Comment by mserour** [comment]
I think it is a bit different because what is happening is some point cloud frames suddenly stop and then continue to publish again. It suddenly freezes on RVIZ and then it works normal again. Therefore, the frame rate drops when trying to ros topic hz the topic it self. Also, we didn't launch the node that publishes the LIDAR image and right now RVIZ is just visualizing the point cloud but the same issue happens that the point cloud freezes from time to time.

**Comment by Samahu** [comment]
> The LIDAR starts to consistently publish the data at 10 FPS, however, with time evolves, the FPS keeps dropping and there is some sort of skittering/discontinuity in the LIDAR visualization from RVIZ.

How soon it takes to observe the drop in the frame rate?

**Comment by mserour** [comment]
I think it is observed immediately, some frames freeze after operation. I am sorry the word of skittering or discontinuity is misleading and incorrect. It is different issue than 110. I think the problem right now is the frames freeze as visualized in RVIZ for some reason and then the average FPS is dropped because you lost some frames.

**Comment by Samahu** [comment]
I carried out few tests around this issue, here are few findings:
- There is a noticeable drop in the frame rate when using the driver with ros-foxy
- It didn't matter whether I have used **FastRTP** or **CycloneDDs**, same behavior (I haven't tested other RMW since they are slightly more involved).
- This behavior is not observed when using the rolling distro on Ubuntu 20.04 machine.. meaning I would receive 10Hz on the `/ouster/points` and the rate is consistent. Also verified it visually, i.e there is not significant drop.
- I briefly tested the driver against humble at some point and don't recall having significant frame drop but another user reported poor performance on humble #63

In short I would say the complaint is valid but it is very distro/architecture/dds dependent. I would be addressing this and other concerns as part of a larger effort that I am working on. I will keep you updated on the issue. In the meantime you may consider experimenting with the rolling/humble if it is rather something that you can possibly consider.
